dissect/target/plugins/os/windows/sam.py

```diff
@@ -137,83 +210,225 @@
         ("uint32", "rid"),
         ("string", "fullname"),
         ("string", "username"),
-        ("string", "comment"),
-        ("datetime", "lockout"),
-        ("datetime", "creation"),
+        ("string", "admincomment"),
+        ("string", "usercomment"),
         ("datetime", "lastlogin"),
+        ("datetime", "lastpasswordset"),
+        ("datetime", "lastincorrectlogin"),
         ("uint32", "flags"),
+        ("uint16", "countrycode"),
         ("uint32", "failedlogins"),
         ("uint32", "logins"),
         ("string", "lm"),
-        ("string", "ntlm"),
+        ("string", "nt"),
     ],
 )
 
 
-class SamPlugin(Plugin):
-    """SAM plugin."""
+def expand_des_key(key: bytes) -> bytes:
+    # Expand the key from a 7-byte password key into an 8-byte DES key
+    key = bytearray(key[:7]).ljust(7, b"\x00")
+    s = bytearray(
+        [
+            ((key[0] >> 1) & 0x7F) << 1,
+            ((key[0] & 0x01) << 6 | ((key[1] >> 2) & 0x3F)) << 1,
+            ((key[1] & 0x03) << 5 | ((key[2] >> 3) & 0x1F)) << 1,
+            ((key[2] & 0x07) << 4 | ((key[3] >> 4) & 0x0F)) << 1,
+            ((key[3] & 0x0F) << 3 | ((key[4] >> 5) & 0x07)) << 1,
+            ((key[4] & 0x1F) << 2 | ((key[5] >> 6) & 0x03)) << 1,
+            ((key[5] & 0x3F) << 1 | ((key[6] >> 7) & 0x01)) << 1,
+            (key[6] & 0x7F) << 1,
+        ]
+    )
+    return bytes(s)
+
 
-    KEY = "HKEY_LOCAL_MACHINE\\SAM\\SAM\\Domains\\Account\\Users"
+def rid_to_key(rid: int) -> tuple[bytes, bytes]:
+    s = rid.to_bytes(4, "little", signed=False)
+    k1 = expand_des_key(bytes([s[0], s[1], s[2], s[3], s[0], s[1], s[2]]))
+    k2 = expand_des_key(bytes([s[3], s[0], s[1], s[2], s[3], s[0], s[1]]))
 
-    def check_compatible(self):
-        if not len(list(self.target.registry.key(self.KEY))) > 0:
-            raise UnsupportedPluginError(f"Registry key not found: {self.KEY}")
+    return k1, k2
+
+
+def aes_decrypt(key: bytes, value: bytes, iv: bytes) -> bytes:
+    plaintext = b""
+    cipher = AES.new(key, AES.MODE_CBC, IV=iv)
+    n = 16
+    for block in [value[i : i + n] for i in range(0, len(value), n)]:
+        plaintext += cipher.decrypt(block)
+
+    return plaintext
```

Comment:
Isn't this the same as:
```python
return AES.new(key, AES.MODE_CBC, iv).decrypt(value)
```

I believe pycryptodome does the block loop automatically.